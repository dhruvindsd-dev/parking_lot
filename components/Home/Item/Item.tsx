import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { LotType } from "../../../context/GlobalState";

const Item = ({
	onPress,
	lotNum,
	startTime,
	carRegistrationNum,
}: LotType & { onPress: () => void }) => {
	return (
		<TouchableOpacity
			testID={
				carRegistrationNum
					? `parking-drawing-registered-${lotNum}`
					: `parking-drawing-space-${lotNum}`
			}
			onPress={onPress}
			style={styles.container}>
			<View
				style={{
					...styles.item,
					backgroundColor: carRegistrationNum ? "#5781FF" : undefined,
				}}>
				<Text
					style={{
						...styles.text,
						color: carRegistrationNum ? "white" : undefined,
					}}>
					{lotNum}
				</Text>
				{carRegistrationNum && (
					<Text
						testID={`parking-drawing-space-number-${lotNum}`}
						style={{ fontSize: 10, color: "white" }}>
						{carRegistrationNum}
					</Text>
				)}
			</View>
		</TouchableOpacity>
	);
};
const styles = StyleSheet.create({
	container: {
		flex: 1,
		maxWidth: "25%",
	},
	item: {
		minHeight: 90,
		padding: 10,
		margin: 5,
		borderRadius: 10,
		borderWidth: 2,
		borderColor: "#5781FF",
		flex: 1,
		backgroundColor: "#F5F7FF",
		height: "100%",
		justifyContent: "center",
		alignItems: "center",

		// box-shadow:  20px 20px 60px #bebebe,
		//  -20px -20px 60px #ffffff;
	},
	text: {
		fontSize: 18,
		fontWeight: "bold",
		color: "#5781FF",
	},
});

export default Item;
