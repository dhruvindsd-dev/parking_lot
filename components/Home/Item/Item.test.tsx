import renderer from "react-test-renderer";
import Item from "./Item";

test("Home Parking Lot Component Renders Correctly", () => {
	const tree = renderer
		.create(
			<Item
				lotNum={1}
				onPress={() => {}}
				carRegistrationNum={"Mh-101"}
				startTime={new Date("12-12-12")}
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
