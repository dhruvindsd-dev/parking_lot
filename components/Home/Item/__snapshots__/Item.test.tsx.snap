// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Parking Lot Component Renders Correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "maxWidth": "25%",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5781FF",
        "borderColor": "#5781FF",
        "borderRadius": 10,
        "borderWidth": 2,
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "margin": 5,
        "minHeight": 90,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 10,
        }
      }
    >
      Mh-101
    </Text>
  </View>
</View>
`;
